### PR TITLE
Unifying the IdealGyrator

### LINKED ISSUE #2786: Unify Gyrator
Currently we have both Modelica.Electrical.Analog.Basic.Gyrator and Modelica.Electrical.Analog.Ideal.IdealGyrator. (I don't see any use of them in MSL.)

To me it would make more sense to unify into the Basic one along the lines of:
```
model Gyrator
...
  parameter SI.Conductance G(start=1) "Gyration conductance";
  parameter SI.Conductance G2=G "Secondary gyration conductance";
```
Similarly as Resistor where we have the advanced possibilities in the model, but it as default behaves in the ideal way without any special settings.

(Might be more similar cases.)

### PR DESCRIPTION
**Pull request by dietmarw**
Fixes #2786 
Also added a test model and conversion test.

### CODE REVIEW

File: Modelica/Electrical/Analog/Basic.mo [hunk 1]
```diff
@@ -649,7 +649,7 @@ the user has to allocate the parameter vector <em>L[6] </em>, since <em>Nv=(N*(N
   model Gyrator "Gyrator"
     extends Interfaces.TwoPort;
     parameter SI.Conductance G1(start=1) "Gyration conductance";
-    parameter SI.Conductance G2(start=1) "Gyration conductance";
+    parameter SI.Conductance G2(start=G1) "Gyration conductance";
```
Comment (beutlich) [inline Modelica/Electrical/Analog/Basic.mo]:
```suggestion
    parameter SI.Conductance G2(start=G1) "Secondary gyration conductance";
```

File: ModelicaTestConversion4.mo [hunk 2]
```diff
@@ -442,6 +442,35 @@ Conversion test for <a href=\"https://github.com/modelica/ModelicaStandardLibrar
 </html>"));
       end Issue2361;
 
+      model issue2786IdealGyrator "Conversion test for #2786"
```
Comment (beutlich) [inline ModelicaTestConversion4.mo]:
```suggestion
      model Issue2786IdealGyrator "Conversion test for #2786"
```

File: ModelicaTestConversion4.mo [hunk 3]
```diff
@@ -442,6 +442,35 @@ Conversion test for <a href=\"https://github.com/modelica/ModelicaStandardLibrar
...
+        connect(v2.p, gyrator.p2) annotation (Line(points={{30,10},{30,20},{10,20},{10,8}}, color={0,0,255}));
+        connect(v2.n, gyrator.n2) annotation (Line(points={{30,-10},{30,-20},{10,-20},{10,-12}}, color={0,0,255}));
+        connect(v2.n,ground2. p) annotation (Line(points={{30,-10},{30,-34}}, color={0,0,255}));
+        annotation (experiment(StopTime=1),
+          Documentation(info="<html>
+<p>
+Conversion test for <a href=\"https://github.com/modelica/ModelicaStandardLibrary/issues/2786\">#2786</a>.
+</p>
+</html>"));
+      end issue2786IdealGyrator;
```
Comment (beutlich) [inline ModelicaTestConversion4.mo]:
```suggestion
      end Issue2786IdealGyrator;
```

**Comment by dietmarw** [comment]
I saw that too, tested and the test model does take the correct value of G2=G1 as start value.

**Comment by christiankral** [comment]
If the default parameter `G2` is changed form 

```modelica
  parameter SI.Conductance G2(start=1) "Secondary gyration conductance";
```

to 

```modelica
  parameter SI.Conductance G2(start=G1) "Secondary gyration conductance";
```

this is a non-backwards compatible change if a user has a model with:

```modelica
 Modelica.Electrical.Analog.Basic.Gyrator gyrator(G1 = 2);
```

This is not a likely application, but I wanted to comment on it. So even though it would make sense to change the default value of `G2` to be backwards compatible with the `Ideal` Gyrator, it breaks the `Basic` Gyrator. 

Is it possible to consider the change of a default value in the conversion script?

**Comment by beutlich** [comment]
@christiankral is right. We must not change the default or start values of Modelica.Electrical.Analog.Basic.Gyrator to not break backwards-compatibility.

The following conversion works in Dymola 2020 (with original Modelica.Electrical.Analog.Basic.Gyrator):

```
convertClass("Modelica.Electrical.Analog.Ideal.IdealGyrator", "Modelica.Electrical.Analog.Basic.Gyrator")
convertModifiers("Modelica.Electrical.Analog.Ideal.IdealGyrator", {"G", "G"}, {"G1.start=%G%", "G2.start=%G%"});
```

Not sure if it the dot-notation of the start attribute is legal Modelica. If it is not (and if we want it), we need to clarify it in the specification as long as there only is one tool supporting conversion scripts.

File: ModelicaTestConversion4.mo [hunk 4]
```diff
@@ -442,6 +442,35 @@ Conversion test for <a href=\"https://github.com/modelica/ModelicaStandardLibrar
 </html>"));
       end Issue2361;
 
+      model Issue2786IdealGyrator "Conversion test for #2786"
+        extends Modelica.Icons.Example;
+        Modelica.Electrical.Analog.Basic.Ground ground1 annotation (Placement(transformation(extent={{-40,-54},{-20,-34}})));
+        Modelica.Electrical.Analog.Ideal.IdealGyrator gyrator(G=1) annotation (Placement(transformation(extent={{-10,-12},{10,8}})));
```
Comment (beutlich) [inline ModelicaTestConversion4.mo]:
```suggestion
        Modelica.Electrical.Analog.Ideal.IdealGyrator gyrator(G=2) annotation (Placement(transformation(extent={{-10,-12},{10,8}})));
```

Would be good to have here a value different from the default start value.

**Comment by beutlich** [comment]
> Not sure if the dot-notation of the start attribute is legal for conversions. If it is not (and if we want it), we need to clarify it in the specification as long as there only is one tool supporting conversion scripts.

The following (with the start attribute in parentheses) works, too:

```
convertClass("Modelica.Electrical.Analog.Ideal.IdealGyrator",
             "Modelica.Electrical.Analog.Basic.Gyrator")
convertModifiers("Modelica.Electrical.Analog.Ideal.IdealGyrator",
                 {"G", "G"}, {"G1(start=%G%)", "G2(start=%G%)"})
```

I believe this might be the preffered conversion.

**Comment by HansOlsson** [comment]
> 
> 
> > Not sure if the dot-notation of the start attribute is legal for conversions. If it is not (and if we want it), we need to clarify it in the specification as long as there only is one tool supporting conversion scripts.
> 
> The following (with the start attribute in parentheses) works, too:
> 
> ```
> convertClass("Modelica.Electrical.Analog.Ideal.IdealGyrator",
>              "Modelica.Electrical.Analog.Basic.Gyrator")
> convertModifiers("Modelica.Electrical.Analog.Ideal.IdealGyrator",
>                  {"G", "G"}, {"G1(start=%G%)", "G2(start=%G%)"})
> ```
> 
> I believe this might be the preferred conversion.

I believe it should be simplified to:
```
convertModifiers("Modelica.Electrical.Analog.Ideal.IdealGyrator",
                 {"G"}, {"G1(start=%G%)", "G2(start=%G%)"})
```

As there is only one old G-parameter

**Comment by beutlich** [comment]
> 
> I believe it should be simplified to:
> 
> ```
> convertModifiers("Modelica.Electrical.Analog.Ideal.IdealGyrator",
>                  {"G"}, {"G1(start=%G%)", "G2(start=%G%)"})
> ```
> 
> As there is only one old G-parameter

Confirmed. Thanks. (I thought both arrays need to be of same size.)

@dietmarw Do you want me to push to your branch to fix the conversion?

**Comment by beutlich** [comment]
And, we also need to keep

```
convertElement({"Modelica.Electrical.Analog.Ideal.IdealGyrator"}, "G", "G1");
```

**Comment by dietmarw** [comment]
@beutlich Feel free to push.

**Comment by dietmarw** [comment]
@christiankral @AHaumer , any chance to give this a quick look so it can be merged?

**Comment by dietmarw** [comment]
Agreed and changed.

**Pull request merged**